File: 03-dfs-bfs-graph/43165/acisliver.js
let answer = 0;

function solution(numbers, target) {

    dfs(numbers, target,0, 0);
    return answer;
}

function dfs(numbers, target, current, count) {
    if (count === numbers.length) {
        if (current === target)
            answer++;
        return;
    }

    dfs(numbers, target, current + numbers[count], count + 1)
    dfs(numbers, target, current - numbers[count], count + 1)

}

File: Leetcode/455/Acisliver.java
class Solution {
    public int findContentChildren(int[] g, int[] s) {
        Arrays.sort(g);
        Arrays.sort(s);

        int count = 0;
        int gI = 0;
        for (int i = 0; i < s.length; i++) {
            if (gI == g.length) {
                break;
            }
            if (s[i] >= g[gI]) {
                count += 1;
                gI += 1;
            }
        }

        return count;
    }
}

